Panic on trying to check fyne

I just got it using `go get` on my Linux computer (with Go 1.15) and tried to run it like `tickeryzer ./...` within the main folder of [fyne](https://github.com/fyne-io/fyne) and received this panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x6b9673]

goroutine 5374 [running]:
github.com/orijtech/tickeryzer.run.func1(0x7a3780, 0xc030b45d00, 0x1, 0xc00081ce00, 0x5, 0x20, 0x1)
	/home/jacob/go/src/github.com/orijtech/tickeryzer/tickeryzer.go:93 +0x293
golang.org/x/tools/go/ast/inspector.(*Inspector).WithStack(0xc035a9caa0, 0xc000eb6b30, 0x1, 0x1, 0xc000eb6b20)
	/home/jacob/go/src/golang.org/x/tools/go/ast/inspector/inspector.go:126 +0x142
github.com/orijtech/tickeryzer.run(0xc02a250f70, 0xc02a250f70, 0x0, 0x0, 0x0)
	/home/jacob/go/src/github.com/orijtech/tickeryzer/tickeryzer.go:55 +0x112
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce(0xc0368ddcc0)
	/home/jacob/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:691 +0x8b9
sync.(*Once).doSlow(0xc0368ddcc0, 0xc0013c3f90)
	/usr/lib64/golang/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
	/usr/lib64/golang/src/sync/once.go:57
golang.org/x/tools/go/analysis/internal/checker.(*action).exec(0xc0368ddcc0)
	/home/jacob/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:579 +0x65
golang.org/x/tools/go/analysis/internal/checker.execAll.func1(0xc0368ddcc0)
	/home/jacob/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:567 +0x34
created by golang.org/x/tools/go/analysis/internal/checker.execAll
	/home/jacob/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:573 +0x125
```

--- a/testdata/src/a/a.go
+++ b/testdata/src/a/a.go
@@ -2,8 +2,11 @@ package p
 
 import (
 	"fmt"
+	"sync"
 	"time"
 )
+
+var m sync.Mutex
 
 func f() {
 	ticker := time.NewTicker(time.Second) // want `missing ticker.Stop\(\) call`
@@ -29,6 +32,16 @@ func h() {
 	}()
 
 	for t := range ticker1.C {
+		fmt.Println("Current time: ", t)
+	}
+}
+
+func i() {
+	ticker := time.NewTicker(time.Second) // want `missing ticker.Stop\(\) call`
+	m.Lock()
+	defer m.Unlock()
+
+	for t := range ticker.C {
 		fmt.Println("Current time: ", t)
 	}
 }

--- a/testdata/src/a/a.go
+++ b/testdata/src/a/a.go
@@ -1,6 +1,7 @@
 package p
 
 import (
+	"context"
 	"fmt"
 	"sync"
 	"time"
@@ -43,6 +44,17 @@ func i() {
 
 	for t := range ticker.C {
 		fmt.Println("Current time: ", t)
+	}
+}
+
+func j(ctx context.Context) {
+	ticker := time.NewTicker(time.Second) // OK
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+		default:
+		}
 	}
 }
 

--- a/tickeryzer.go
+++ b/tickeryzer.go
@@ -84,15 +84,18 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		}
 
 		for _, stmt := range stmts[1:] {
+			var root *ast.Ident
 			switch stmt := stmt.(type) {
 			case *ast.DeferStmt:
-				if root := rootIdent(stmt.Call.Fun); root.Obj == ticker.Obj {
-					return true
-				}
+				root = rootIdent(stmt.Call.Fun)
 			case *ast.ExprStmt:
-				if root := rootIdent(stmt.X); root.Obj == ticker.Obj {
-					return true
-				}
+				root = rootIdent(stmt.X)
+			}
+			if root == nil {
+				continue
+			}
+			if root.Obj == ticker.Obj {
+				return true
 			}
 		}
 

--- a/tickeryzer.go
+++ b/tickeryzer.go
@@ -83,26 +83,49 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return true // could not find the *time.Ticker in the assignment.
 		}
 
-		for _, stmt := range stmts[1:] {
-			var root *ast.Ident
-			switch stmt := stmt.(type) {
-			case *ast.DeferStmt:
-				root = rootIdent(stmt.Call.Fun)
-			case *ast.ExprStmt:
-				root = rootIdent(stmt.X)
-			}
-			if root == nil {
-				continue
-			}
-			if root.Obj == ticker.Obj {
-				return true
-			}
+		if hasTickerStopCall(pass, ticker, stmts[1:]) {
+			return true
 		}
 
 		pass.ReportRangef(ticker, "missing %s.Stop() call", ticker.Name)
 		return true
 	})
 	return nil, nil
+}
+
+func hasTickerStopCall(pass *analysis.Pass, ticker *ast.Ident, stmts []ast.Stmt) bool {
+	for _, stmt := range stmts {
+		var root *ast.Ident
+		switch stmt := stmt.(type) {
+		case *ast.DeferStmt:
+			root = rootIdent(stmt.Call.Fun)
+		case *ast.ExprStmt:
+			root = rootIdent(stmt.X)
+		case *ast.ForStmt:
+			if hasTickerStopCall(pass, ticker, stmt.Body.List) {
+				return true
+			}
+		case *ast.SelectStmt:
+			if hasTickerStopCall(pass, ticker, stmt.Body.List) {
+				return true
+			}
+		case *ast.CommClause:
+			if hasTickerStopCall(pass, ticker, stmt.Body) {
+				return true
+			}
+		case *ast.CaseClause:
+			if hasTickerStopCall(pass, ticker, stmt.Body) {
+				return true
+			}
+		}
+		if root == nil {
+			continue
+		}
+		if root.Obj == ticker.Obj {
+			return true
+		}
+	}
+	return false
 }
 
 func isTimeNewTicker(info *types.Info, expr *ast.CallExpr) bool {
@@ -144,6 +167,8 @@ func rootIdent(n ast.Node) *ast.Ident {
 	switch n := n.(type) {
 	case *ast.SelectorExpr:
 		return rootIdent(n.X)
+	case *ast.CallExpr:
+		return rootIdent(n.Fun)
 	case *ast.Ident:
 		return n
 	default:

